ts/IterableInbox.tsx

```diff
@@ -22,46 +23,59 @@ const IterableInbox = ({
 }: inboxProps) => {
    const defaultInboxTitle = "Inbox"
    const [isDisplayMessage, setIsDisplayMessage] = useState<boolean>(false)
-   const [selectedMessageId, setSelectedMessageId] = useState<string>("")
-   const [messages, setMessages] = useState<InboxRowViewModel[]>([])
+   const [selectedMessageIdx, setSelectedMessageIdx] = useState<number>(0)
+   const [rowViewModels, setRowViewModels] = useState<InboxRowViewModel[]>([])
+
    const inboxDataModel = new IterableInboxDataModel()
+   // const inboxDeleteSource = new IterableInAppDeleteSource()
 
    const fetchData = async () => {
-      let newMessages = await inboxDataModel.refresh()
+      let newRowViewModels = await inboxDataModel.refresh()
 
-      newMessages = newMessages.map((message, index) => {
-         return {...message, last: index === newMessages.length - 1}
+      newRowViewModels = newRowViewModels.map((rowViewModel, index) => {
+         return {...rowViewModel, last: index === newRowViewModels.length - 1}
       })
 
-      setMessages(newMessages)
+      setRowViewModels(newRowViewModels)
    }
 
    useEffect(() => {
       fetchData()
    }, [])
 
-   const selectedMessage = messages.find(message => message.inAppMessage.messageId === selectedMessageId)
+   const selectedMessage = rowViewModels[selectedMessageIdx]
 
-   function handleMessageSelect(id: string, index: number, messages: InboxRowViewModel[]) {
-      let newMessages = messages.map((message) => {
-         return (message.inAppMessage.messageId === id) ?
-            {...message, read: true } : message
+   function handleMessageSelect(id: string, index: number, rowViewModels: InboxRowViewModel[]) {
+      let newRowViewModels = rowViewModels.map((rowViewModel) => {
+         return (rowViewModel.inAppMessage.messageId === id) ?
+            {...rowViewModel, read: true } : rowViewModel
       })
-      setMessages(newMessages)
+      setRowViewModels(newRowViewModels)
       inboxDataModel.setItemAsRead(index)
 
       setIsDisplayMessage(true)
-      setSelectedMessageId(id)
+      setSelectedMessageIdx(index)
    }
 
-   // function deleteMessage(id: string, index: number, messages: InboxRowViewModel[]) {
-   //    let newMessages = messages.filter((message) => {
-   //       return message.inAppMessage.messageId !== id
-   //    })
-   //    inboxDataModel.deleteItem(index, inboxSwipe)
-   //    //newMessages[newMessages.length - 1] = {...newMessages[newMessages.length - 1], last: true}
-   //    setMessages(newMessages)
-   // }
+   // const onDelete = (id: string, index: number, rowViewModels: InboxRowViewModel[]) => {useCallback(event => {
+   //    deleteMessage(id, index, rowViewModels)
+   // }, rowViewModels)}
+
+   const deleteMessage = async (id: string, index: number, rowViewModels: InboxRowViewModel[]) => {
```

Comment:
I'm not sure how necessary these comments around here are, but they're very confusing in here. They should either be removed, or another comment saying why they're here should be put in.

The `deleteMessage` function should probably be renamed to something else, it's not clear what a `Message` is. I would suggest `deleteRow` to keep the notion that this is an inbox with a bunch of rows.

Also, this function should not need both the `index` (which should be a `row` to prevent confusion), and `id` (which should be the `messageId` or 'inAppMessageId`), `index` isn't used anywhere in here, including in the commented code, and the rest of the components now have to utilize this signature.

Reply:
Renamed deleteMessage function to deleteRow. The id parameter is renamed to messageId.